Core/Systems/Questing/Quests/TestQuest/TestQuestTwo.cs

```diff
@@ -0,0 +1,35 @@
+﻿using PathOfTerraria.Content.Items.Gear.Armor.Leggings;
+using PathOfTerraria.Core.Systems.ModPlayers;
+using PathOfTerraria.Core.Systems.Questing.QuestStepTypes;
+using PathOfTerraria.Core.Systems.Questing.RewardTypes;
+using System.Collections.Generic;
+using PathOfTerraria.Content.Items.Gear.Weapons.Sword;
+using Terraria.ID;
+
+namespace PathOfTerraria.Core.Systems.Questing.Quests.TestQuest;
+
+internal class TestQuestTwo : Quest
+{
+	public override QuestTypes QuestType => QuestTypes.MainStoryQuestAct1;
+	public override string Name => "Test Quest 2";
+	public override string Description => "This is another test quest. Simply used for testing purposes";
+
+	protected override List<QuestStep> _subQuests =>
+	[
+		new CollectCount(ItemID.StoneBlock, 50, s => $"Collect {s} stone."),
+		new KillCount(x => x.lifeMax > 100, 10, remaining => $"Kill {remaining} mobs with 100+ max life")
+	];
+
+	public override int NPCQuestGiver => throw new NotImplementedException();
```

Comment:
For futureproofing reasons, we should default to using -1 or another valid but clear value that is not going to instantly crash, but it also clearly unused. This is a test quest, so I doubt this'll be kept enough to go to major quest expansion, much less production, but still.

Reply:
Would rather that we will never have to not give it a value, therefore it's marked as abstract; so you need to override it. 
But we don't use it yet so there's no reason to give it a value yet.

Though honestly it might be better to not have this variable at all, you might start with a quest or get it from an item or something else there might also be multiple NPCs involved - just added it since the issue said to.

Reply:
We could do something like the `IEntitySource` system where we track arbitrarily what assigned the quest. Not super sure what it'd be used for, but that'd be way more consistent than an NPC type only. We'd also be able to define a default that doesn't give any info but also doesn't crash, yeah. That does make sense.

Reply:
Sure, but I don't really get why we have to know what/who gave/will give the quest, my assumption with the NPCQuestGiver was that you would tell the NPC that they could give this quest to the player, but that feels unnecessary and problematic in multiple ways.

Reply:
What ways would that be? For Spirit we did the same thing, if only to be like 
```
"The Final Task"
from the Adventurer
```
so it wasn't a big deal, but it also wasn't an issue ever. That being said, it's not really important at all either.

Reply:
If you want a quest line start at npc a and come back to npc a in some future quest, multiplier ways to get one specific quest, locking quests - can only take over of the 3 (probably a bad example), none of these really are; many it's not a problem, just feel like if you had quests linked to dialogue optics it'd be more versatile... 🤷‍♂️

Reply:
Oh you mean storing the NPC instance, yeah that'd definitely cause issues. This only tracks type, which is effectively constant in terms of gameplay, but yeah. That's definitely something we need to be wary of.
What do you mean "dialogue optics" exactly?

Reply:
Nothing about instances; edited it to dialogue options.

Reply:
Now I have no idea what you're trying to say lol sorry

Reply:
I don't think I do either...

Clean slate: 
I think it would work well and be a versatile system if quest were created/added by the thing that gives the quest using some method call, if it's an npc, call the method when a certain dialogue options is picked or for an item, when it's used/consumed.

Instead of the quest telling the NPC that the NPC can give the quest to the player, or that the quest tells the item, but that sure wouldn't make sense at all - that would just be an "empty" item of a certain type, and that seems really dumb...
Or something like that, the thought is maby a little too vague...

Reply:
> I think it would work well and be a versatile system if quest were created/added by the thing that gives the quest using some method call, if it's an npc, call the method when a certain dialogue options is picked or for an item, when it's used/consumed.

Is this not the case already? At the moment, Blacksmith gives you his quest by talking to him, and making any other system would be a ton more work for no reason. We haven't done anything like an empty item for a quest or anything.

Reply:
Well, should have just said my assumption was wrong, then, lol

> my assumption with the NPCQuestGiver was that you would tell the NPC that they could give this quest to the player, but that feels unnecessary and problematic in multiple ways

Reply:
Didn't even know we had a quest giving npc
xD

Reply:
But dose the blacksmiths need and or use this variable, if not can't we just remove it?

Reply:
Well, it'd just be for `"Forging a New Blade" from the Blacksmith`, it's not necessary but it's also not going to cause any issues. I don't care either way but I don't see why we'd need to remove it.

> my assumption with the NPCQuestGiver was that you would tell the NPC that they could give this quest to the player, but that feels unnecessary and problematic in multiple ways

I didn't understand this honestly, my bad. Misinterpreted it lol

Reply:
Oh... For the title... I think I'm the one that misinterpreted the dumbest, so de.

I feel like quests normally don't have the giver like this in games, not like I have that many games to bare that on, though. - or many it's more that the pool of genres I play is small.

But if it's just that then I don't care either way, either.

Reply:
No worries lol. Yeah it's super inconsequential, we could remove it or change it super easily in the future if we want to.

Reply:
Updated to -1